### PR TITLE
Audit the `option` and `ptr` types for proper integer usage

### PR DESCRIPTION
**Pull request by nikomatsakis**
cc https://github.com/rust-lang/rust/issues/22240

### CODE REVIEW

File: src/libcore/option.rs [hunk 1]
```diff
@@ -188,10 +188,10 @@ impl<T> Option<T> {
     /// # Example
     ///
     /// ```
-    /// let x: Option<uint> = Some(2);
+    /// let x: Option<usize> = Some(2);
     /// assert_eq!(x.is_some(), true);
     ///
-    /// let x: Option<uint> = None;
+    /// let x: Option<usize> = None;
```
Comment (Gankro) [inline src/libcore/option.rs]: Is there a reason to leave these as `usize` rather than `i32`? (for pedagogy against using usize randomly)

Comment (Gankro) [inline src/libcore/option.rs]: Some of these can actually be inferred if we want i32, although I dunno if that would "mess up" the quality of the docs.

**Comment by huonw** [comment]
Looks good, although there are a lot of examples using `usize` unnecessarily; it seems the only one that needs it is the example calling `.len` on a `&String`.

r=me all except that one are changed to `u32` or something.

File: src/libcore/option.rs (hunk 1 above)
Comment (nikomatsakis) [inline src/libcore/option.rs]: I figured the explicit types were there for a reason (in particular, type inference is hard on the reader, and these docs are largely targeting total newbies). In that case, I don't know that it matters what type is there. But I guess I can change it to `u32` or `i32`.

**Comment by nikomatsakis** [comment]
@huonw updated the option examples and threw on a few more modules.

**Pull request merged**